### PR TITLE
Documentation, add more examples

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Add more `runnableExamples` to documentation, only Documentation.

### CODE REVIEW

File: lib/system/assertions.nim [hunk 1]
```diff
@@ -50,15 +50,15 @@ template assert*(cond: untyped, msg = "") =
   ## `command line switches <nimc.html#compiler-usage-command-line-switches>`_.
   ##
   ## .. code-block:: nim
-  ##   static: assert 1 == 9, "This works when not built with -d:danger or --assertions:off"
+  ##   static: assert 1 == 9, "This assertion passes when not built with -d:danger or --assertions:off"
```
Comment (Clyybber) [inline lib/system/assertions.nim]: But it doesn't "pass". It triggers because 1 != 9..

File: lib/pure/htmlgen.nim [hunk 2]
```diff
@@ -808,3 +808,102 @@ runnableExamples:
       annotation(encoding = "application/x-tex", title = "Latex on Web", r"x^{2} + y")
     )
   ) == """<math><semantics><annotation encoding="application/x-tex" title="Latex on Web">x^{2} + y</annotation></semantics></math>"""
+
+
+runnableExamples:
```
Comment (Araq) [inline lib/pure/htmlgen.nim]: Come on, that's not an "example", that's a unit test and nobody likes to read it.

File: lib/pure/terminal.nim [hunk 3]
```diff
@@ -908,6 +908,40 @@ proc newTerminal(): owned(PTerminal) =
   when defined(windows):
     initTerminal(result)
 
+
+runnableExamples:
```
Comment (Araq) [inline lib/pure/terminal.nim]: Same, this is not example code, that's a test. It also shows implementation details...

File: lib/std/varints.nim [hunk 4]
```diff
@@ -120,6 +120,55 @@ proc decodeZigzag*(x: uint64): int64 {.inline.} =
   let casted = cast[int64](x)
   result = (`shr`(casted, 1)) xor (-(casted and 1))
 
+
+runnableExamples:
```
Comment (Araq) [inline lib/std/varints.nim]: Same.

File: lib/pure/concurrency/cpuinfo.nim [hunk 5]
```diff
@@ -98,3 +95,8 @@ proc countProcessors*(): int {.rtl, extern: "ncpi$1".} =
   else:
     result = sysconf(SC_NPROCESSORS_ONLN)
   if result <= 0: result = 0
+
+
+runnableExamples:
+  block:
```
Comment (Araq) [inline lib/pure/concurrency/cpuinfo.nim:101]: The next time please avoid pointless ``block`` statements.

**Pull request merged**